switchbot/__init__.py

```diff
@@ -141,6 +188,25 @@ def press(self) -> bool:
         """Press command to device."""
         return self._sendcommand(PRESS_KEY, self._retry_count)
 
+    def switch_mode(self) -> str:
+        """Return Toggle or Switch from cache.
+        Run update first."""
+        if self._mode != 0:
+            return "switch"
+        return "toggle"
+
+    def is_on(self) -> bool:
+        """Return switch state from cache.
+        Run update first."""
+        if self._is_on != 0:
```

Comment:
I think it is better to move this logic to the update function

Reply:
Done.